core/arch/arm/kernel/generic_entry_a32.S

```diff
@@ -432,7 +445,15 @@ UNWIND(	.cantunwind)
 	bl	get_core_pos
 	bl	cpu_is_ready
 
+#if defined (CFG_BOOT_SECONDARY_REQUEST)
+	/* get ns entry from ns_entry_addrs array */
+	bl  get_core_pos
+	ldr r1, =ns_entry_addrs
+	add r1, r1, r0, lsl #2	/* paddr_t is 4 bytes in 32-bit system */
```

Comment:
The size of `paddr_t` is not guaranteed to be 4 bytes on a 32-bit system. It would be better to use the `uint32_t` type instead for `ns_entry_addrs[]` as anything larger than that doesn't make sense on a 32-bit system.